lib/pry/input_completer.rb

```diff
@@ -239,4 +227,39 @@ def build_path(input)
     end
     return path, input
   end
+
+  private
+
+  def self.fetch_obj_space_collection
```

Comment:
what do you mean by "on every switch" ? the binding is received in Pry::InputCompleter#call, and that wouldn't change. `#initialize` accepts an input object(like Readline for example) and an instance of Pry.

the only difference between global and instance-local state would be that we would store the collection and count as instance variables on an instance of Pry::InputCompleter, and avoid the mutex and class-level state entirely. nothing else would change, even how bindings are passed into Pry::InputCompleter would not change.